src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/events/ReferralEventListener.kt

```diff
@@ -0,0 +1,45 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.ApplicationListener
+import org.springframework.stereotype.Component
+import org.springframework.web.util.UriComponentsBuilder
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.NotifyService
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.SNSService
+
+@Component
+class ReferralEventListener(
```

Comment:
i've consolidated all the event side effects here so it's easier to see what is being triggered.